### PR TITLE
Remove unused includes of boost/assert.hpp from implementation

### PR DESCRIPTION
**Pull request by BenjaminW3**

### CODE REVIEW

**Comment by BenjaminW3** [comment]
You are right. I will have a look at those.

**Comment by BenjaminW3** [comment]
I have replaced them with exceptions in the same way we already did in the OpenMP accelerators.

**Comment by ax3l** [comment]
PR needs rebase :)

**Comment by BenjaminW3** [comment]
rebased

**Comment by BenjaminW3** [comment]
push

File: include/alpaka/exec/ExecGpuCudaRt.hpp [hunk 1]
```diff
@@ -282,22 +304,8 @@ namespace alpaka
...
-                        reinterpret_cast<unsigned int *>(&gridDim)[i] = static_cast<unsigned int>(gridBlockExtent[TDim::value-1u-i]);
-                        reinterpret_cast<unsigned int *>(&blockDim)[i] = static_cast<unsigned int>(blockThreadExtent[TDim::value-1u-i]);
-
-                    }
-                    // Assert that all extent of the higher dimensions are 1!
-                    for(auto i(std::min(static_cast<typename TDim::value_type>(3), TDim::value)); i<TDim::value; ++i)
-                    {
-                        BOOST_VERIFY(gridBlockExtent[TDim::value-1u-i] == 1);
-                        BOOST_VERIFY(blockThreadExtent[TDim::value-1u-i] == 1);
-                        BOOST_VERIFY(threadElemExtent[TDim::value-1u-i] == 1);
```
Comment (psychocoderHPC) [inline include/alpaka/exec/ExecGpuCudaRt.hpp]: Could it be that `threadElemExtent` not checked any more because of the refactoring?

Comment (BenjaminW3) [reply include/alpaka/exec/ExecGpuCudaRt.hpp]: You are right, thank you for reviewing this. I will fix it.

Comment (BenjaminW3) [reply include/alpaka/exec/ExecGpuCudaRt.hpp]: It should be fixed now!

**Pull request merged**